packages/menu/src/Menu.ts

```diff
@@ -125,6 +198,77 @@ export class Menu extends SpectrumElement {
         this.removeEventListener('keydown', this.handleKeydown);
     }
 
+    public async selectOrToggleItem(item: MenuItem): Promise<void> {
+        const resolvedSelects = this.resolvedSelects;
+        if (resolvedSelects == null) {
+            return;
+        }
+
+        const oldSelectedItemsMap = new Map(this.selectedItemsMap);
+        const oldSelected = this.selected.slice();
+        const oldSelectedItems = this.selectedItems.slice();
+        const oldValue = this.value;
+
+        if (resolvedSelects === 'single') {
+            this.selectedItemsMap.clear();
+            this.selectedItemsMap.set(item, true);
+            this.value = item.value;
+            this.selected = [item.value];
+            this.selectedItems = [item];
+        } else {
+            if (this.selectedItemsMap.has(item)) {
+                this.selectedItemsMap.delete(item);
+            } else {
+                this.selectedItemsMap.set(item, true);
+            }
+
+            // Match HTML select and set the first selected
+            // item as the value. Also set the selected array
+            // in the order of the menu items.
+            let valueSet: boolean = false;
+            let selected: string[] = [];
+            let selectedItems: MenuItem[] = [];
+            for (const menuItem of this.menuItems) {
+                if (this.selectedItemsMap.has(menuItem)) {
+                    if (!valueSet) {
+                        this.value = menuItem.value;
+                        valueSet = true;
+                    }
+                    selected.push(item.value);
+                    selectedItems.push(item);
+                }
+            }
+            this.selected = selected;
+            this.selectedItems = selectedItems;
+            this.value = this.selected.join(this.valueSeparator);
+        }
+
+        await this.updateComplete;
+        const applyDefault = this.dispatchEvent(
+            new Event('change', {
+                cancelable: true,
+                bubbles: true,
+            })
+        );
+        if (!applyDefault) {
+            this.selected = oldSelected;
+            this.selectedItems = oldSelectedItems;
+            this.selectedItemsMap = oldSelectedItemsMap;
+            this.value = oldValue;
+            return;
+        }
+        if (resolvedSelects === 'single') {
+            for (const oldItem of oldSelectedItemsMap.keys()) {
+                if (oldItem !== item) {
+                    oldItem.selected = false;
+                }
+            }
+            item.selected = true;
+        } else if (resolvedSelects === 'multiple') {
+            item.selected = !item.selected;
+        }
```

Comment:
I don't fully grok what is happening here. Can you comment a little for next me, who is unequivocally going to be less smart than current me.